Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -102,33 +104,54 @@ public Task<bool> IsCompleteSubmissionAsync(string code)
             return false;
         }
 
-        public override Task<LspResponse> LspMethod(string methodName, JObject request)
+        public override async Task<LspResponse> LspMethod(string methodName, JObject request)
```

Comment:
Can we use a more specific, made-for-purpose type for the request?

Reply:
We can't because each request is a JSON object that's unique to each request type, so there's no deserialization we can do ahead of time that's meaningful.

Reply:
The other alternative would be to strongly-type methods and their requests, but that would force each kernel to implement something like this:

``` C#
override TextDocumentHoverResponse Hover(HoverParams hoverParams) { }
override TextDocumentCompletionResponse Completion(CompletionParams completion) { }
override TextDocumentSignatureHelpResponse SignatureHelp(SignatureHelpParams sigHelp() { }
// ... there are a ridiculous amount of methods that we'd have to add/override
```

The other alternative I considered is adding `ILspKernel` with the above methods using default interface methods to simply return `null`.

Reply:
It's generally a good idea to keep this layer ignorant of serialization. (The `Newtonsoft.Json` reference exists only for the sake of `JsonFormatter` and `Microsoft.DotNet.Interactive.Server`, not for core functionality.)

So the latter approach will be more flexible over time. I would go even further, in fact, and decompose into one interface per method, because they won't all make sense for all kernels. Something like this?

```csharp
public interface ISupportHoverText
{
    TextDocumentHoverResponse Hover(HoverParams hoverParams);
}
public interface ISupportCompletion
{
    TextDocumentCompletionResponse Completion(CompletionParams completion);
}
public interface ISupportSignatureHelp
{
    TextDocumentSignatureHelpResponse SignatureHelp(SignatureHelpParams sigHelp();
}
```

Reply:
(but of course F# can't implement an interface with default methods, yet)

Reply:
Done.